File: excel/jython/update/excel_update.py
#! /usr/bin/jython
# -*- coding: utf-8 -*-
#
#	excel_update.py
#					Aug/08/2013
#
# ----------------------------------------------------------------
import	sys
import	string
#
#
# ----------------------------------------------------------------
sys.path.append ('/var/www/data_base/common/python_common')
sys.path.append ('/var/www/data_base/common/jython_common')
from text_manipulate import dict_update_proc
from jython_text_manipulate import jython_dict_display_proc
from jython_excel_manipulate import jython_excel_read_proc
from jython_excel_manipulate import jython_excel_write_proc
# ----------------------------------------------------------------
print ("*** 開始 ***")
#
excel_file = sys.argv[1]
key = sys.argv[2]
population_in = string.atoi (sys.argv[3])
print ("%s\t%d" % (key, population_in))
#
dict_aa = jython_excel_read_proc (excel_file)
#
if key in dict_aa:
	dict_aa=dict_update_proc (dict_aa,key,population_in)
	jython_dict_display_proc (dict_aa)
	jython_excel_write_proc (excel_file,dict_aa)
#
#
print ("*** 終了 ***")
#
# ----------------------------------------------------------------

File: excel/jruby/update/excel_update.rb
#! /usr/bin/jruby
# -*- coding: utf-8 -*-
#
#	excel_update.rb
#					Aug/08/2013
#
# --------------------------------------------------------
load '/var/www/data_base/common/ruby_common/text_manipulate.rb'
load '/var/www/data_base/common/jruby_common/excel_manipulate.rb'
# --------------------------------------------------------
puts "*** 開始 ***"
xlsx_file=ARGV[0]
key = ARGV[1]
population_in = ARGV[2].to_i
puts key,population_in
#
dict_aa = excel_read_proc(xlsx_file)

if (dict_aa.key?(key))
	dict_aa[key]["population"] = population_in
	dict_aa[key]["date_mod"] = Date.today.to_s
	dict_display_proc(dict_aa)
	excel_write_proc(xlsx_file,dict_aa)
end
#
puts "*** 終了 ***"
# --------------------------------------------------------
